tests/v2/validation/snapshot/snapshot.go

```diff
@@ -327,10 +314,10 @@ func snapshotV2Prov(t *testing.T, client *rancher.Client, podTemplate corev1.Pod
 		}
 	}
 
-	return initialSnapshotConfig{initialKubernetesVersion, initialControlPlaneConcurrencyValue, initialWorkerConcurrencyValue, snapshotToRestore}
+	return restoreSnapshotConfig, initialSnapshotConfig{initialKubernetesVersion, initialControlPlaneConcurrencyValue, initialWorkerConcurrencyValue, snapshotToRestore}
```

Comment:
same question here about returning the custom object now that we also return the whole cluster

Reply:
L238 is an example of why this is still here. We want to make sure that the current values match what the value is prior to restoring a snapshot.

This especially is important when we're doing upgrade scenarios.

Reply:
Discussed offline, resolved this.